cql2pgjson/src/main/java/org/folio/cql2pgjson/CQL2PgJSON.java

```diff
@@ -612,11 +614,13 @@ private String getWrapTemplateWithSchemaDetection(String whereField,Table target
    * @param targetTable  where to search
    * @return the ForeignKeys if found, null otherwise
    */
-  private ForeignKeys findForeignKey(String field, Table targetTable) {
-    String fieldName = dbTable.getTableName() + field;
-    for (ForeignKeys key : targetTable.getForeignKeys()) {
-      if (fieldName.equalsIgnoreCase(key.getFieldName())) {
-        return key;
+  private ForeignKeys findForeignKey(String field, Table targetTable, Table currentTable) {
+    String fieldName = currentTable.getTableName() + field;
+      if(targetTable.getForeignKeys() != null) {
```

Comment:
Can you make it a guard clause that reduces the indentation of the following block and therefore improves readability?

    if (targetTable.getForeignKeys() == null) {
      return null;
    }
    for (ForeignKeys key : …